Sources/RNLibreToolsiOS13/LibreToolsResources/Sensor.swift

```diff
@@ -482,17 +482,43 @@ class Sensor: ObservableObject {
         return crcReport.contains("FAILED") && history.count > 0
     }
 
-    func detailFRAM() throws {
+    struct sensorInfo {
+        var type: SensorType
+        var family: SensorFamily
+        var region: SensorRegion
+        var serial: String
+        var state: String
+        var lastReadingDate: Date
+        var age: Int
+        var maxLife: Int
+        var initializations: Int
+    }
+
+    func detailFRAM() -> AnyObject {
+
+        let response = sensorInfo(
+            type: type,
+            family: family,
+            region: region,
+            serial: serial,
+            state: state.description,
+            lastReadingDate: lastReadingDate,
+            age: age,
+            maxLife: maxLife,
+            initializations: initializations
+        )
+
+
         if encryptedFram.count > 0 && fram.count >= 344 {
             logger.info("\(fram.hexDump(header: "Sensor decrypted FRAM:", startBlock: 0))")
         }
-        
-        if crcReport.count > 0 {
```

Comment:
I think we still need to validate CRC and throw error. If CRC validation failed, data is corrupted and it's risky to return corrupted data, especially for a medical product

Reply:
How can I do that? @yury-dymov